src/Camera.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+** EPITECH PROJECT, 2022
+** indieStudio
+** File description:
+** Camera
+*/
+
+#include "Camera.hpp"
+
+#include "raylib.h"
+#include <memory>
+#include <iostream>
+
+/* #ifndef DEBUG
+    #define DEBUG
+    #include <iostream>
+#endif */
+
+static int getNewId(void)
+{
+    static int idCounter = 0;
+
+    return (idCounter++);
+}
+
+namespace indie {
+Camera::Camera(Vector3 target, Vector3 position)
+{
+    Vector3 up = {0.0f, 1.0f, 0.0f};
+    _camera = std::make_shared <Camera3D>();
+    _camera->target = target;
+    std::cout << "target: " << _camera->target.x << " " << _camera->target.y << " " << _camera->target.z << std::endl;
+    _camera->position = position;
+    std::cout << "position: " << _camera->position.x << " " << _camera->position.y << " " << _camera->position.z << std::endl;
+    _camera->up = up;
+    _camera->fovy = 45.0f;
+    _camera->projection = CAMERA_PERSPECTIVE;
+    _id = getNewId();
+    #ifdef DEBUG
+        std::cerr << "\tIndiCam::Camera(Vector3 target, Vector3 position) called" << std::endl;
+    #endif
+}
+
+Camera::Camera(std::shared_ptr<Vector3> target, std::shared_ptr<Vector3> position)
+: _targetSnap(target), _positionSnap(position), _snapMode(true)
+{
+    Vector3 up = {10.0f, 3.0f, 0.0f};
```

Comment:
maybe it would be good if it were a bit clearer . for example, explaining that those number are the coordinate x,y,z.